### PR TITLE
Fix builder image version dropdown value in pipeline edit flow

### PR DESCRIPTION
**Pull request by karthikjeeyar**
**Fixes**: 

https://issues.redhat.com/browse/ODC-5536

**Analysis / Root cause**: 

In pipeline add/edit git flow, builder image version was not tracked in the newly created pipeline, hence the previously selected version was getting lost while going through the edit flow.

**Solution Description**: 

Added a label `pipeline.openshift.io/runtime-version` to get and set the builder image version and used it in the edit flow.

**Screen shots / Gifs for design review**: 

version-edit-flow

**Unit test coverage report**: 

  ```
updatePipelineForImportFlow
    ✓ should dissociate pipeline if template is not available (1ms)
    ✓ should update params if template is of same type
    ✓ should update VERSION params if image tag is changed (1ms)
    ✓ should update pipeline if template is of different type (1ms)
  isDockerPipeline
    ✓ should return false for a non docker based pipelines
    ✓ should return true for a docker based pipeline template
  pipelineRuntimOrVersionChanged
    ✓ should return false if runtime and version is same for two given pipelines
    ✓ should return false if runtime or version is different for two given pipelines
```
**Test setup:**

1. In git add flow, choose a builder image and version
2. check `Add pipeline` option
3. create the resource and choose edit option from topology page.
4. edit form should have pre-selected the correct builder image version in the dropdown 

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Review by debsmita1 — CHANGES_REQUESTED** [review]

File: frontend/packages/pipelines-plugin/src/components/import/pipeline/pipeline-template-utils.ts [hunk 1]
```diff
@@ -76,7 +80,13 @@ export const createPipelineForImportFlow = async (
   template.metadata = {
     name: `${name}`,
     namespace,
-    labels: { ...template.metadata?.labels, 'app.kubernetes.io/instance': name },
+    labels: {
+      ...template.metadata?.labels,
+      'app.kubernetes.io/instance': name,
+      ...(template?.metadata?.labels?.[PIPELINE_STRATEGY_LABEL] !== 'docker' && {
```
[the remark this answers is not in the document]
Comment (karthikjeeyar) [reply frontend/packages/pipelines-plugin/src/components/import/pipeline/pipeline-template-utils.ts]: I missed it there. Updated :+1:

File: frontend/packages/pipelines-plugin/src/components/import/pipeline/pipeline-template-utils.ts [hunk 2]
```diff
@@ -61,6 +65,18 @@ export const getPipelineParams = (
   });
 };
 
+export const pipelineRuntimOrVersionChanged = (
```
Comment (debsmita1) [inline frontend/packages/pipelines-plugin/src/components/import/pipeline/pipeline-template-utils.ts:68]:
```suggestion
export const pipelineRuntimeOrVersionChanged = (
```

**Review by debsmita1 — CHANGES_REQUESTED** [review]

**Comment by debsmita1** [comment]
@karthikjeeyar 5 tests are failing

**Comment by karthikjeeyar** [comment]
@debsmita1 fixed it

**Comment by debsmita1** [comment]
Try:
Import from catalog (DONOT check the pipeline) -> Edit the app -> change the BI version & check the pipeline & submit the form
 -> Edit the form --- I do not see the updated version selected

**Comment by karthikjeeyar** [comment]
> Try:
> Import from catalog (DONOT check the pipeline) -> Edit the app -> change the BI version & check the pipeline & submit the form
> -> Edit the form --- I do not see the updated version selected

@debsmita1  Ah I see what is happening, first time creation (without Pipeline) creates Buildconfig, then while editing the application if you check pipeline even then it takes buildconfig's selected version over the pipelines. The precedence should be Pipelines -> Buildconfig -> default fallback.

**Comment by karthikjeeyar** [comment]
> Seems to work fine, but I noticed something in the edit flow as well, is this desired?

Not sure what you are referring here, I tried using a `php` image it detected with right builder image and version.

**Comment by abhinandan13jan** [comment]
> > Seems to work fine, but I noticed something in the edit flow as well, is this desired?
> 
> Not sure what you are referring here, I tried using a `php` image it detected with right builder image and version.

I understand this is separate from the Bug, but the above warning `unable to detect the builder image` in Edit flow, is it desired?

**Comment by abhinandan13jan** [comment]
PR is fine on its own. Checked locally.
/LGTM
@karthikjeeyar could you please log a tracker for the warning issue we see.

**Comment by karthikjeeyar** [comment]
> > > Seems to work fine, but I noticed something in the edit flow as well, is this desired?
> > 
> > 
> > Not sure what you are referring here, I tried using a `php` image it detected with right builder image and version.
> 
> I understand this is separate from the Bug, but the above warning `unable to detect the builder image` in Edit flow, is it desired?

I can reproduce this using dotnet image `https://github.com/redhat-developer/s2i-dotnetcore-ex.git`, this one is NOT desired it shouldn't show warning if the builder image is selected in edit flow.

**Comment by abhinandan13jan** [comment]
https://issues.redhat.com/browse/ODC-5568 cc: @karthikjeeyar

**Review by jerolimov — APPROVED** [review]
Tested it locally. Works fine. :+1: 

/approve

**Comment by karthikjeeyar** [comment]
cherry-pick release-4.7

**Pull request merged**